### ISSUE TITLE
Invalid video icon state when entering room

### ISSUE THREAD

**Issue by everbslab** [twilio/twilio-video-app-react#220]
**Describe the bug**
When user joins the video room we are seeing that Camera Control Icon does not match whether camera is actually on/off for participant. It shows as "Camera is Off", but the control icon shows that video is enabled. The same appears while using mobile devices.

Another case: 
When I entered chat room first time, camera was off but camera icon was incorrect. Screenshot is attached. It was maybe because Chrome asked to confirm access to camera and mic but after confirmation nothing changed.
In all next cases of opening videochat (all except first) camera is on by default and camera icon is correct.

**To Reproduce**
1. Enter the room created
2. Scroll down to Controls
4. See error video icon show that videocamera is enabled but no stream flows

**Expected behavior**
Camera icon in the controls should show state of camera stream enabled and vice versa.

**Screenshots**
![image](https://user-images.githubusercontent.com/970419/83240514-b2485180-a1c3-11ea-8de1-3a5d01ee3f47.png)

**Environment (please complete the following information):**
 - OS: Windows any version, MacOS, iOS 13.5
 - Browser: Chrome > 80, Safari 13.1
 - App Version: master branch of this repository
 - SDK Version: 2.5.0
 - Node.js version: 12.14

**Comment by timmydoza** [comment]
Thanks for the issue @everbslab!

I tried to reproduce the issue that you described and I was not able to. The camera icon appears to function correctly in the master branch of the video app.

Are you running a modified version of the app? The screenshots you provided look different.  If so, could you please provide some code examples, or a link to your repository? 

Thanks!

**Comment by everbslab** [comment]
Hi @timmydoza, 

I've just some style modifications. All others came out-of-the-box.
Sent you invite to private repo for code review. Thank you in advance!

**Comment by timmydoza** [comment]
Thanks for sharing the code @everbslab! I was able to consistently reproduce the issue you described. 

It appears that the problem has to do with the fact that a user's local video track is not published to the room if the user grants audio & video permissions after they join a room. This issue has been reported by another user too: https://github.com/twilio/twilio-video-app-react/issues/201

I'd like to prioritize this issue. I'm hoping to get this fixed some time this week or next. Would that work with your timeline?

**Comment by timmydoza** [comment]
Good to hear @everbslab!

I went ahead and created a PR (https://github.com/twilio/twilio-video-app-react/pull/231) to fix this problem in the React App. I've discussed this issue with the team. We feel that the best solution here is to simply prevent the user from joining the room until the browser's permission dialog has been accepted or denied.

I noticed that your app will join a room automatically, and there is no Join Room button to disable if the device permissions dialog is still pending. I would recommend adding a Join Room button to your app to prevent the user from joining the room if they haven't allowed or denied permissions. 

There's another reason to have a Join Room button. Browsers have an autoplay policy which requires a user interaction (such as clicking on the page) before audio elements can start to play sound. When a users joins a room immediately, audio elements are added to the DOM before there has been any user interaction, and thus no sound can be heard. Adding a Join Room button is one way to prevent problems with the autoplay policy, because it allows audio elements to be added to the page after a user interaction, in which case they will play audio without issue. 

Let me know if you have any additional questions!
